### PR TITLE
[New Rule] Cisco Secure Endpoint Malicious File Detection

### LINKED ISSUE #2878: [New Rule] Cisco Secure Endpoint Malicious Threat Detected Event
## Description
The idea of this rule would be to alert Information Security analysts when Cisco Secure Endpoint detects a malicious file. From here, the analyst should investigate to ensure the quarantine event was successful.

### Target indexes
logs-cisco_secure_endpoint.event-*

### Additional requirements
cisco_secure_endpoint Integration

### Target Operating Systems
All

### Query
event.category: malware and event.action: "threat detected" and cisco.secure_endpoint.file.disposition: Malicious

### New fields required in ECS/data sources for this rule?
No

## Example Data
```
{
  "agent": {
    "name": "EventCollector1",
    "id": "cdcd8072-1d30-414a-aa71-0123456789ab",
    "type": "filebeat",
    "ephemeral_id": "e58cd5a2-1d47-4b89-b490-ba9876543210",
    "version": "8.8.1"
  },
  "process": {
    "name": "msedge.exe",
    "pid": 15216,
    "hash": {
      "sha1": "5af110bea6b9688167a46a7f6505c7f6b718a75a",
      "sha256": "a142768daa9a22ae9d9aebbf5c9be4deeef80d1a65db8840606a948fced46e69",
      "md5": "5bc717c58deafdae88094ad737a8f4fa"
    }
  },
  "elastic_agent": {
    "id": "cdcd8072-1d30-414a-aa71-0123456789ab",
    "version": "8.8.1",
    "snapshot": false
  },
  "tags": [
    "cisco-secure_endpoint",
    "forwarded",
    "preserve_original_event"
  ],
  "input": {
    "type": "httpjson"
  },
  "@timestamp": "2023-06-23T11:39:30.000Z",
  "file": {
    "path": "\\\\?\\C:\\Users\\jdoe\\Downloads\\driver-hub-install__28.exe",
    "name": "driver-hub-install__28.exe",
    "hash": {
      "sha1": "77ccf75074599fc076f89060f257feeda5607d33",
      "sha256": "a1f6905b424b2e1479dc823688f3eaffddd8c7537abe9c5ada4a1bcbca25c79c",
      "md5": "ff25f4db981a5980797d736f97adaab6"
    }
  },
  "ecs": {
    "version": "8.8.0"
  },
  "related": {
    "hosts": [
      "Computer1.Contoso.com",
      "Computer1"
    ],
    "ip": [
      "81.2.69.142",
      "10.10.10.10"
    ],
    "user": [
      "jdoe"
    ],
    "hash": [
      "A1F6905B424B2E1479DC823688F3EAFFDDD8C7537ABE9C5ADA4A1BCBCA25C79C",
      "FF25F4DB981A5980797D736F97ADAAB6",
      "77CCF75074599FC076F89060F257FEEDA5607D33",
      "5BC717C58DEAFDAE88094AD737A8F4FA",
      "5AF110BEA6B9688167A46A7F6505C7F6B718A75A",
      "A142768DAA9A22AE9D9AEBBF5C9BE4DEEEF80D1A65DB8840606A948FCED46E69"
    ],
    "mac": [
      "AB:CD:EF:01:23:45"
    ]
  },
  "data_stream": {
    "namespace": "default",
    "type": "logs",
    "dataset": "cisco_secure_endpoint.event"
  },
  "host": {
    "geo": {
      "continent_name": "Europe",
      "region_iso_code": "NL-FL",
      "city_name": "Almere Stad",
      "country_iso_code": "NL",
      "country_name": "Netherlands",
      "region_name": "Flevoland",
      "location": {
        "lon": 5.1753,
        "lat": 52.2192
      }
    },
    "hostname": "Computer1",
    "as": {
      "number": 29396,
      "organization": {
        "name": "Eurofiber Nederland BV"
      }
    },
    "ip_public": "81.2.69.142",
    "os": {
      "family": "windows",
      "platform": "windows"
    },
    "domain": "contoso.com",
    "ip": [
      "10.10.10.10"
    ],
    "name": "Computer1",
    "mac": [
      "AB:CD:EF:01:23:45"
    ]
  },
  "threat": {
    "technique": {
      "id": [
        "T1105",
        "T1204",
        "T1204.003",
        "T1569"
      ]
    },
    "tactic": {
      "id": [
        "TA0002",
        "TA0011",
        "TA0042"
      ]
    }
  },
  "event": {
    "severity": 2,
    "agent_id_status": "verified",
    "ingested": "2023-06-23T11:39:44Z",
    "original": "{\"data\":{\"computer\":{\"active\":true,\"connector_guid\":\"76543210-abcd-efab-fedc-0123456789ab\",\"external_ip\":\"81.2.69.142\",\"hostname\":\"Computer1.contoso.com\",\"links\":{\"computer\":\"https://api.amp.cisco.com/v1/computers/76543210-abcd-efab-fedc-0123456789ab\",\"group\":\"https://api.amp.cisco.com/v1/groups/76543210-89ab-cdef-fedc-0123456789ab\",\"trajectory\":\"https://api.amp.cisco.com/v1/computers/76543210-abcd-efab-fedc-0123456789ab/trajectory\"},\"network_addresses\":[{\"ip\":\"10.10.10.10\",\"mac\":\"AB:CD:EF:01:23:45\"}],\"user\":\"jdoe@Contoso\"},\"connector_guid\":\"76543210-abcd-efab-fedc-0123456789ab\",\"date\":\"2023-06-23T11:39:30+00:00\",\"detection\":\"W32.A1F6905B42-95.SBX.TG\",\"detection_id\":\"7247844800483819523\",\"event_type\":\"Threat Detected\",\"event_type_id\":1090519054,\"file\":{\"disposition\":\"Malicious\",\"file_name\":\"driver-hub-install__28.exe\",\"file_path\":\"\\\\\\\\?\\\\C:\\\\Users\\\\jdoe\\\\Downloads\\\\driver-hub-install__28.exe\",\"identity\":{\"md5\":\"ff25f4db981a5980797d736f97adaab6\",\"sha1\":\"77ccf75074599fc076f89060f257feeda5607d33\",\"sha256\":\"a1f6905b424b2e1479dc823688f3eaffddd8c7537abe9c5ada4a1bcbca25c79c\"},\"parent\":{\"disposition\":\"Unknown\",\"file_name\":\"msedge.exe\",\"identity\":{\"md5\":\"5bc717c58deafdae88094ad737a8f4fa\",\"sha1\":\"5af110bea6b9688167a46a7f6505c7f6b718a75a\",\"sha256\":\"a142768daa9a22ae9d9aebbf5c9be4deeef80d1a65db8840606a948fced46e69\"},\"process_id\":15216}},\"group_guids\":[\"76543210-89ab-cdef-fedc-0123456789ab\"],\"id\":7247844800483820000,\"severity\":\"Medium\",\"tactics\":[\"TA0002\",\"TA0011\",\"TA0042\"],\"techniques\":[\"T1105\",\"T1204\",\"T1204.003\",\"T1569\"],\"timestamp\":1687520370,\"timestamp_nanoseconds\":473000000},\"metadata\":{\"links\":{\"self\":\"https://api.amp.cisco.com/v1/events?limit=100\\u0026offset=0\\u0026start_date=2023-06-23T11%3A37%3A18%2B00%3A00\"},\"results\":{\"current_item_count\":12,\"index\":0,\"items_per_page\":100,\"total\":12}},\"version\":\"v1.2.0\"}",
    "code": "1090519054",
    "created": "2023-06-23T11:39:41.809Z",
    "kind": "alert",
    "action": "threat detected",
    "id": "7247844800483820000",
    "category": [
      "file",
      "malware"
    ],
    "dataset": "cisco_secure_endpoint.event"
  },
  "cisco": {
    "secure_endpoint": {
      "detection": "W32.A1F6905B42-95.SBX.TG",
      "group_guids": [
        "76543210-89ab-cdef-fedc-0123456789ab"
      ],
      "event_type_id": 1090519054,
      "computer": {
        "network_addresses": [
          {
            "ip": "10.10.10.10",
            "mac": "AB:CD:EF:01:23:45"
          }
        ],
        "connector_guid": "76543210-abcd-efab-fedc-0123456789ab",
        "active": true,
        "external_ip": "81.2.69.142"
      },
      "file": {
        "parent": {
          "disposition": "Unknown"
        },
        "disposition": "Malicious"
      },
      "connector_guid": "76543210-abcd-efab-fedc-0123456789ab",
      "related": {
        "mac": [
          "AB:CD:EF:01:23:45"
        ]
      },
      "detection_id": "7247844800483819523"
    }
  },
  "user": {
    "domain": "Contoso",
    "name": "jdoe"
  }
}
```

### PR DESCRIPTION
**Pull request by MakoWish**
## Issues
- Closes #2878 

## Summary
This detection rule queries for events indicating Cisco Secure Endpoint detected a malicious file.

## Contributor checklist

- [X] Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- [X] Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

**Comment by MakoWish** [comment]
Test says the field `cisco.secure_endpoint.file.disposition` is unknown, but you can see the field exists in the sample data in: #2878

**Comment by terrancedejesus** [comment]
👋🏽 Thanks for creating this rule!

Here is what needs to be done for unit tests to pass.
1. run `python -m detection_rules dev integrations build-manifests -o -i cisco_secure_endpoint` (adds `cisco_secure_endpoint` manifests to the `integration-manifests` file)
2. run `python -m detection_rules dev integrations build-schemas -o` (will add the `cisco_secure_endpoint` schema to the `integration-schemas` file)

Once this is done, the rule validation workflow should be able to find the integration specific fields in the integration schema, thus validating they exist.

**Comment by MakoWish** [comment]
Conflicts are too complex? Never seen that before. Did I miss something? Or is that fixable on your end?

**Comment by terrancedejesus** [comment]
> Conflicts are too complex? Never seen that before. Did I miss something? Or is that fixable on your end?

Rather odd. Did you by chance pull in all updates upstream into your fork before adjusting the manifest and schema files?

**Comment by MakoWish** [comment]
I reverted back, updated my fork and branch from main, then I ran the commands to update the schemas and manifests, but it is still failing for the same field.

File: rules/integrations/cisco_secure_endpoint/cse_malicious_file_detection.toml [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+"""
+risk_score = 99
+rule_id = "637c4b00-1214-11ee-980f-adf4cd9645cd"
+severity = "critical"
+tags = ["Domain: Endpoint", "Data Source: Cisco Secure Endpoint", "Use Case: Threat Detection", "Tactic: Execution", "Tactic: Command and Control"]
+timestamp_override = "event.ingested"
+type = "query"
+
+query = '''
+event.category: malware and event.action: "threat detected" and cisco.secure_endpoint.file.disposition: Malicious
```
Comment (terrancedejesus) [inline rules/integrations/cisco_secure_endpoint/cse_malicious_file_detection.toml:73]:
```suggestion
event.category: malware and event.dataset: "cisco_secure_endpoint.event" and event.action: "threat detected" and cisco.secure_endpoint.file.disposition: Malicious
```

Whenever your data is based on a data stream from an integration, `event.dataset` has to be set.
Reference: https://github.com/elastic/integrations/blob/f9ef14cb1f6aa48a20eb60553ed1e02dd01561a7/packages/cisco_secure_endpoint/data_stream/event/fields/base-fields.yml#L20C5-L20C6

We have this error else where, but we should add it earlier.

Comment (MakoWish) [reply rules/integrations/cisco_secure_endpoint/cse_malicious_file_detection.toml]: Do you know if there are plans to set `event.dataset` on all Integrations? I can see it is done here, but as it is now, `data_stream.dataset` seems to be the standard, and there are quite a lot where `event.dataset` is not utilized. 

I finally got this to pass locally. Just pushed the changes.

**Review by terrancedejesus — APPROVED** [review]
PR looks good, thanks for the addition!

**Comment by MakoWish** [comment]
Not sure how to fix those conflicts now. I synchronized from main, checked out the manifest and schema files from `origin`, then rebuilt both locally, tested, and pushed. Tried twice, but still conflicting? Got any tips on how to resolve that?

**Comment by MakoWish** [comment]
@terrancedejesus, any idea how to fix those conflicts?

**Pull request closed without merging**